nilearn/maskers/nifti_masker.py

```diff
@@ -95,6 +95,14 @@ def _filter_and_mask(
     # as small as possible in order to speed up the process
 
     if not _utils.niimg_conversions._check_same_fov(imgs, mask_img_):
+        warnings.warn(
+            'imgs are being resampled to the mask_img resolution. '
+            'This process is memory intensive when using a '
+            'high resolution mask. You might want to provide '
+            'a target_affine or resample the mask beforehand '
+            'to save memory and computation time.',
```

Comment:
```suggestion
            'imgs are being resampled to the mask_img resolution. '
            'This process is memory intensive. You might want to provide '
            'a target_affine that is equal to the affine of the imgs '
            'or resample the mask beforehand '
            'to save memory and computation time.',
```

I think the point of this message is to avoid resampling, whether up or down, which is why I removed the "high resolution mask" part. Indeed using high resolution data in an analysis pipeline is memory intensive in general but it's not the point of this warning. I added that the target affine should be equal to the data affine so that the mask is resampled to this. So that would produce the same behavior as resampling the mask to the data beforehand.

See relevant part of code where the mask would be resampled which then avoids the behavior of resampling the data that comes later in the process: https://github.com/nilearn/nilearn/blob/2e18c91342b65048307a21a45b5a8473414b6d16/nilearn/maskers/nifti_masker.py#L471-L477